docs/dev/guide/contract-verification.md

```diff
@@ -0,0 +1,86 @@
+# Contract Verification
+
+## What is contract verification?
+
+The bytecode generated by the Solidity compiler is essentially deployed when a Smart Contract is deployed on a network. The original Solidity code is not retained in any way.
+
+Verifying a contract entails determining whether the on-chain bytecode generated from the Smart Contract source code complies. If it does, we can state that we have confirmed the bytecode's source code's integrity. Hence the name – verification.
+<br>
+
+<hr>
+
+<br>
+
+## How are Smart Contracts verified in zkSync??
+
+The deployed bytecode and the Solidity source code of a Smart Contract are compared during the verification process using zkSync. The algorithm compiles the source code to compare the generated bytecode to the deployed one.
+<br>
+If the two sides match in every way, the contract is verified.
+
+Code verification requires six arguments: 
+
+- The compiler address
+- The contract name
+- The source code, including the source of all imports
+- The version of the compiler used to produce the deployed bytecode
+- Information about the number of compiler optimizations (if any)
+- The constructor arguments
+ 
+ If any of these pieces of information is wrong, the process of verification fails.
+<br>
+
+<hr>
+
+<br>
+
+## Modes of verification
+
+Depending on whether your contract is deployed on a network or with zkSync, there are two modes of contract verification:
+
+- Public: Every zkSync user on the block explorer will be able to check the verified source code and interact with it. 
+- Private: The contract is verified and thus valid within a single project in zkSync, available to you and anybody with access to the project.
+
+
+
+## Verifying Contracts Using the zkSync Block Explorer
+
+You can verify your contract so that zkSync can convert it all into a format that is readable by everyone. It is beneficial if you want to make the most of our zkEVM Debugger capabilities.
+
+To get started click the **Tools** tab in the top header, a dropdown will pop up and select **Smart Contract Verification**,  after which you will be shown the following screen:
+
+![Smart Contract Verification page!](../../assets/images/verify-contract.png "verify contact")
+<br>
+
+<hr>
+
+<br>
+
+### Enter contract details
+
+To verify the contract, enter the following details;
+
+- Contract address: The address provided must match with `0x` address supplied during contract creation.
+- Contract name: The name must be identical to the name provided in the contract.
+- Optimization: This checks whether or not optimization was used while compiling the contract, If you enabled optimization during compilation, check **Yes** otherwise select **No**
+- Compiler version: This specifies the exact version of the compiler used to compile a Smart Contract. Click on the dropdown to specify the compiler version used.
```

Comment:
we need to explain somewhere how to obtain the compiler version

Reply:
<img width="614" alt="Screenshot 2022-08-26 at 13 34 44" src="https://user-images.githubusercontent.com/14915525/186904462-46e7575c-94e7-4824-b51d-383eb91258a3.png">

I'm not sure how to go about this.

From your comment, you said "the explorer would offer to choose the compiler version", how then can users programmatically obtain the compiler version asides from what the explorer provides?

Reply:
These tasks are completed by now, so there's no need to go in depth about it (what you have there already seems fine to me).

My comment was that we need to specify to devs to be aware of the compiler version they use while compiling their contracts, so they would know which version to use on the block explorer.

Once this is completed, we should add some text about how to retrieve it from artifacts:
https://linear.app/matterlabs/issue/SMA-1199/add-compiler-version-to-artifacts